src/functions/pattern.ts

```diff
@@ -1,5 +1,25 @@
 import { IFunction, IFunctionResult, IRulePatternOptions } from '../types';
 
+function test(value: string, regex: any) {
+  let re;
+  if (typeof regex === 'string') {
+    // regex in a string like {"match": "/[a-b]+/im"} or {"match": "[a-b]+"} in a json ruleset
+    // the available flags are "gimsuy" as described here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp
+    const splitRegex = /^\/(.+)\/([gimsuy]*)$/.exec(regex);
+    if (splitRegex) {
+      // with slashes like /[a-b]+/ and possibly with flags like /[a-b]+/im
+      re = new RegExp(splitRegex[1], splitRegex[2]);
```

Comment:
Syntax error caused by `/foo/iii` is uncaught here. Shall we wrap it with try...catch?
```suggestion
      try {
        re = new RegExp(splitRegex[1], splitRegex[2]);
      } catch {
        return false;
      }
```

Reply:
By silently catching errors, users wouldn't be aware that their regex is invalid. I would prefer to keep a behavior similar to what happens when you provide an invalid schema to the schema function (the exception is thrown, that really helped me 😅 ).
The following output shows how it looks when the schema and pattern function throw an exception:
```
Adding OpenAPI 2.0 (Swagger) functions
Encountered error when running rule 'invalid-regex-modifiers' on node at path '$,definitions':
SyntaxError: Invalid flags supplied to RegExp constructor 'izorglub'
Encountered error when running rule 'test-schema-error' on node at path '$,definitions,ValidSchemaName':
Error: schema is invalid: data/type should be equal to one of the allowed values, data/type should be array, data/type should match some schema in anyOf

/Users/arnaud/Dev/spectral-testing/samples/pattern-swagger.yaml
 23:24  error  schema-name-is-valid-not-working      Definitionname must not end with dto (any case)
 23:24  error  schema-name-is-valid-ulgy-workaround  Definitionname must not end with dto (any case)

✖ 2 problems (2 errors, 0 warnings, 0 infos)
```
If you're OK with that I just need to update the test to modify the one with an invalid flag. Can I do that on my branch and then use the `yarn commit` command without doing another pull request?

Reply:
> By silently catching errors, users wouldn't be aware that their regex is invalid. I would prefer to keep a behavior similar to what happens when you provide an invalid schema to the schema function (the exception is thrown, that really helped me sweat_smile ).

Agreed.

> If you're OK with that I just need to update the test to modify the one with an invalid flag. Can I do that on my branch and then use the yarn commit command without doing another pull request?

I'm totally fine with that. You can do it on your branch, sure.